### ISSUE TITLE
Replace all (or part of) the default links failed

### ISSUE THREAD

**Issue by RachelCT1** [vasturiano/3d-force-graph#478]
I want to use custom curve links to replace all (or part of) the default links . Is there any method to update the default links? I have tried add my custom links like this:
`let material = new THREE.LineBasicMaterial({color: 0x929E5F});
      let curveGroup = new THREE.Group();
      bundleRes.forEach((d, idx) => {
          let vectors = [];
          d.map((s,i) => {vectors.push(new THREE.Vector3(s.x, s.y, s.z));});
          let points = new THREE.CatmullRomCurve3(vectors).getPoints(50);
          let curve = new THREE.Line(new THREE.BufferGeometry().setFromPoints(points), material);
          curveGroup.add(curve);
      });
savedGraph[0].scene().add(curveGroup);`
but how can I remove the default links? I have tried to get the default links, and dispose them, but it seems didn't work.

**Comment by vasturiano** [comment]
@RachelCT1 have you tried setting `.linkThreeObjectExtend(false)`?

**Comment by RachelCT1** [comment]
> @RachelCT1 have you tried setting `.linkThreeObjectExtend(false)`?

yes. I have tried, but I am not sure that I use right. I tried like this:
`

let material = new THREE.LineBasicMaterial({color: 0x929E5F});

let curveGroup = new THREE.Group();

bundleRes.forEach((d, idx) => {
let vectors = [];
d.map((s,i) => {vectors.push(new THREE.Vector3(s.x, s.y, s.z));});
let points = new THREE.CatmullRomCurve3(vectors).getPoints(50);
let curve = new THREE.Line(new THREE.BufferGeometry().setFromPoints(points), material);
curveGroup.add(curve);
});

Graph.scene().add(curveGroup);
Graph.linkThreeObjectExtend(false);

`
The network display both default links and custom links (I ant remove these default purple links).
![image](https://user-images.githubusercontent.com/36122719/124136226-2506b900-dab7-11eb-9269-daf3519ebca4.png)

I tried the second one:
`

let material = new THREE.LineBasicMaterial({color: 0x929E5F});

//curveGroup is array
let curveGroup = [];

bundleRes.forEach((d, idx) => {
let vectors = [];
d.map((s,i) => {vectors.push(new THREE.Vector3(s.x, s.y, s.z));});
let points = new THREE.CatmullRomCurve3(vectors).getPoints(50);
let curve = new THREE.Line(new THREE.BufferGeometry().setFromPoints(points), material);
curveGroup.push(curve);
});

Graph.scene().add(curveGroup);
Graph.linkThreeObjectExtend(false)
 .linkThreeObject(link=>{
        return curveGroup[link.index];
      });;


`
but what I get is like this, the green links. The defeat links are purple, and my custom links are green and curve, so I am confused.
![image](https://user-images.githubusercontent.com/36122719/124136468-65fecd80-dab7-11eb-8e8d-c9649c66e859.png)

**Comment by vasturiano** [comment]
@RachelCT1 I noticed you're adding objects to the scene directly, outside of the component's control:
```js
Graph.scene().add(curveGroup);
```

Unless you are really adding objects unrelated to force-graph, I'd suggest to use the `linkThreeObject` and `nodeThreeObject` to implement custom links/nodes.

**Comment by opheliagame** [comment]
@RachelCT1 hey, I have a similar issue, and I was wondering what `bundleRes` refers to in your code, and could you maybe give an example of how to construct it
